homeassistant/components/command_line/switch.py

```diff
@@ -65,6 +68,10 @@ async def async_setup_platform(
     switches = []
 
     for object_id, device_config in devices.items():
+        _config = {CONF_UNIQUE_ID: device_config.get(CONF_UNIQUE_ID)}
+        _config[CONF_NAME] = Template(
+            device_config.get(CONF_FRIENDLY_NAME, object_id), hass
+        )
```

Comment:
Update the schema instead:
```diff
-        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+        vol.Optional(CONF_FRIENDLY_NAME): cv.template,
```

Reply:
Doesn't work as it uses `object_id` as fallback which is not a template and as name is used in `__init__` it would not have initiated it as Templates`
Removing `object_id` as fallback would be a breaking change